spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -290,7 +293,12 @@ public String toString() {
   private class BatchAppend extends BaseBatchWrite {
     @Override
     public void commit(WriterCommitMessage[] messages) {
-      AppendFiles append = table.newAppend();
+      AppendFiles append;
+      if (branch != null) {
+        append = table.newAppend().toBranch(branch);
+      } else {
+        append = table.newAppend();
+      }
```

Comment:
If we do what I mentioned above we don't need if/else here. A branch will always be defined.